gammapy/spectrum/flux_point.py

```diff
@@ -1426,7 +1432,8 @@ def plot_spectrum(self, ax=None, fp_kwargs=None, model_kwargs=None):
         plot_kwargs.setdefault("color", ax.lines[-1].get_color())
         del plot_kwargs["label"]
 
-        if self.model.parameters.covariance is not None:
+        # TODO: remove this plot error option when there is a better manipulation of uncertainties
+        if self.model.parameters.covariance is not None and plot_error:
```

Comment:
@JouvinLea Maybe just go with a try except statement here and remove the `plot_error` option? Like so:

```
#TODO: remove try except....
try:
    self.model.plot_error()
except WhatEverError:
    log.debug("Model does not support evaluation of errors")
    pass

```